WordPress/gutenberg
# Image: Improve focus management in lightbox

## What?

This PR makes the focus management more consistent in the image lightbox by removing the reference to `event.pointerType` and the differentiation in behavior when using a mouse versus a keyboard.

## Why?

Addresses https://github.com/WordPress/gutenberg/issues/55414 and improves on changes made in https://github.com/WordPress/gutenberg/pull/55212.

Altogether, the previous approach was unreliable, so this PR aims to fix that.

## How?

It removes the logic using `event.pointerType`, and instead...
1.  Makes sure focus is always set to the close button when the lightbox opens.
2. Adds a `tabindex` of -1 to the image, which prevents the focus ring from erroneously showing around the close button when the lightbox is opened in Chrome and Firefox (the focus ring appears to always be visible in this case in desktop Safari, which appears to be the intended browser behavior).
3. Stores a reference to the trigger button so that we can set focus to it when the lightbox closes.

## Testing Instructions




1. Add an image to a post
3. Enable 'Expand on Click' in the image's block settings
4. Publish and view the post
5. Hover over the image and see that the cursor becomes a magnifying glass and that a button appears in the top right corner
6. Click on the image or the button to open the lightbox

On keyboard, test that tabbing to the image will show the lightbox trigger button on focus and that pressing `Enter` or `Spacebar` opens the lightbox as expected.

While the lightbox is open, test that...
1. Pressing `Escape` closes the lightbox
2. Pressing `Tab` will move focus to the close button and trap it
3. That pressing `Enter` or `Spacebar` while on the close button will close the lightbox
4. Clicking on the modal or the close button will close the lightbox

When the lightbox closes, make sure focus is set on the lightbox trigger.
